pages/requests/[uuid].js

```diff
@@ -42,11 +42,14 @@ const Request = () => {
   const { messages, isLoadingMessages, isMessagesError, mutateMessages, messagesData } = useMessages(uuid, accessToken)
   const { files, isLoadingFiles, isFilesError, mutateFiles, filesData } = useFiles(uuid, accessToken)
 
+  const [show, setShow] = useState(false)
+  const [message, setMessage] = useState(null)
   const [allPOs, setAllPOs] = useState([])
   const [isPOError, setIsPOError] = useState(false)
   const [isLoadingPOs, setIsLoadingPOs] = useState(true)
+
   useEffect(() => {
-    if (request) {
+    if (isLoadingPOs && request) {
```

Comment:
currently the `getAllPOs` function below is called nonstop for 2 reasons. this is the **second** reason.

every time the page is refreshed, `request` is still true so we are calling the api and "updating" state, which causes a rerender. (see attached video) 

https://user-images.githubusercontent.com/29032869/228001306-9da479e5-254a-4644-bb80-8e98dd2b9f8c.mp4